### PR TITLE
Add web program to fetch top 10 real time billionaires using the forbes API.

### PR DESCRIPTION
**Pull request by girisagar46**
### Describe your change:
Using the Forbes API to get the top ten real-time billionaires and present in a [rich](https://github.com/Textualize/rich) table.

Here's the sample output:
output

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: web_programming/get_top_billioners.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import tabulate
+
+API_URL = (
+    "https://www.forbes.com/forbesapi/person/rtb/0/-estWorthPrev/true.json"
+    "?fields=personName,gender,source,countryOfCitizenship,"
+    "birthDate,finalWorth&limit=10"
+)
+
+
+def real_time_billionaires():
```
[the remark this answers is not in the document]
Comment (girisagar46) [reply web_programming/get_top_billioners.py]: Provided here: e9baf8f

**Comment by cclauss** [comment]
I like this one.  Instead of Tabulate, would you be willing to do the [table in Rich](https://rich.readthedocs.io/en/stable/tables.html) instead?  I want to use Rich to do some other color-in-the-terminal things and would rather add just one import instead of two.

File: web_programming/get_top_billioners.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        final_response.append(
+            {
+                "Name": person["personName"],
+                "Source": person["source"],
+                "Country": person["countryOfCitizenship"],
+                "Gender": person["gender"],
+                "Worth": f"{person['finalWorth'] / 1000:.1f} Billion",
+                "Age": today.year
+                - birthdate.year
+                - ((today.month, today.day) < (birthdate.month, birthdate.day)),
```
Comment (CaedenPH) [inline web_programming/get_top_billioners.py:32]:
```suggestion
                "Age":  (
                    today.year - birthdate.year - 
                   ((today.month, today.day) < (birthdate.month, birthdate.day))
                ),
```

File: web_programming/get_top_billioners.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    "https://www.forbes.com/forbesapi/person/rtb/0/-estWorthPrev/true.json"
+    "?fields=personName,gender,source,countryOfCitizenship,"
+    "birthDate,finalWorth&limit=10"
+)
+
+
+def real_time_billionaires() -> str:
+    """Get top 10 realtime billionaires using forbes API
+    Returns:
+    Top 10 realtime billionaires date in tabulated string.
```
Comment (CaedenPH) [inline web_programming/get_top_billioners.py:16]:
```suggestion

    Returns the Top 10 realtime billionaires date in tabulated string.
```

File: web_programming/get_top_billioners.py (hunk 2 above)
Comment (cclauss) [reply web_programming/get_top_billioners.py:32]: I believe that psf/black will reformat this back to the original solution (terms on separate lines).

For `rich`, we may need to convert age to a str.

File: web_programming/get_top_billioners.py (hunk 3 above)
Comment (cclauss) [reply web_programming/get_top_billioners.py:16]: This is a comment syntax that some editors (PyCharm, I think) prefer so we should leave it as is.

File: web_programming/get_top_billioners.py (hunk 2 above)
Comment (girisagar46) [reply web_programming/get_top_billioners.py]: Thank you. This has been modified here: c85adac

File: web_programming/get_top_billioners.py (hunk 3 above)
Comment (girisagar46) [reply web_programming/get_top_billioners.py]: Yeah, this was automatically done by PyCharm. However, I've modified it a bit. c85adac

**Comment by girisagar46** [comment]
> I like this one. Instead of Tabulate, would you be willing to do the [table in Rich](https://rich.readthedocs.io/en/stable/tables.html) instead? 

Thank you, @cclauss for reviewing it. Rich is awesome. I've modified it here: c85adac

File: web_programming/get_top_billioners.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
+from datetime import datetime
+
+import requests
+from rich import box
+from rich import console as rich_console
+from rich import table as rich_table
```
Comment (CaedenPH) [inline web_programming/get_top_billioners.py:6]:
```suggestion
from rich import (box, console as rich_console, table as rich_table)
```

**Review by CaedenPH — COMMENTED** [review]
Small change

**Comment by CaedenPH** [comment]
Oh, apparently the `pre-commit` automatically reverts it. No problem, @cclauss, can we get your review?

**Comment by cclauss** [comment]
It does not work on my machine.  I get `json.decoder.JSONDecodeError: Expecting value: line 2 column 1 (char 1)`

But if I go to the URL with my browser, it looks like all the right stuff.

**Comment by cclauss** [comment]
Does it work on your machine @CaedenPH ?

File: web_programming/get_top_billioners.py (hunk 4 above)
Comment (cclauss) [reply web_programming/get_top_billioners.py:6]: [isort](https://pypi.org/project/isort/) sorts your imports so you don't have to.  ;-)

**Comment by CaedenPH** [comment]
> Does it work on your machine @CaedenPH ?

I also get the `JSONDecodeError`

**Comment by cclauss** [comment]
Often we can solve these issues with [`fake_useragent`](https://github.com/TheAlgorithms/Python/search?q=fake_useragent) but that did not fix it for me.

**Comment by girisagar46** [comment]
> Often we can solve these issues with [`fake_useragent`](https://github.com/TheAlgorithms/Python/search?q=fake_useragent) but that did not fix it for me.

Oh!! That's weird. It's working as expected on my mac. 
Can you try by passing `headers = {'Accept': 'application/json'}` on `requests.get` to increase the likelihood of receiving a JSON response. @cclauss

**Comment by cclauss** [comment]
`headers = {'Accept': 'application/json'}` did NOT fix it.

Do you have a Forbes login that you are using?

**Comment by girisagar46** [comment]
> Do you have a Forbes login that you are using?

No, I am not using it. Hitting the API in Incognito mode is giving me the correct response. 
I've changed the API path to `https://www.forbes.com/forbesapi/person/rtb/0/position/true.json`  in this commit e05ef42. 
Can you try it again? @cclauss

**Comment by CaedenPH** [comment]
> > Do you have a Forbes login that you are using?
> 
> No, I am not using it. Hitting the API in Incognito mode is giving me the correct response. I've changed the API path to `https://www.forbes.com/forbesapi/person/rtb/0/position/true.json` in this commit [e05ef42](https://github.com/TheAlgorithms/Python/commit/e05ef42e303d5bca810320d05d1a5060924996a0). Can you try it again? @cclauss

I can hit the api successfully when I visit the url manually, but `requests` automates this process, which triggers forbes's anti-bot protection software, blocking us.

**Comment by cclauss** [comment]
@dhruvmanila Any ideas on how to make this work?  Should we use `httpx` instead of `requests`?  Other ideas?

**Review by cclauss — APPROVED** [review]

File: web_programming/get_top_billioners.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+LIMIT = 10
+TODAY = datetime.now()
+
+API_URL = (
+    "https://www.forbes.com/forbesapi/person/rtb/0/position/true.json"
+    "?fields=personName,gender,source,countryOfCitizenship,birthDate,finalWorth"
+    f"&limit={LIMIT}"
+)
```
Comment (dhruvmanila) [inline web_programming/get_top_billioners.py:19]: `requests` can automatically encode the URL parameters: https://requests.readthedocs.io/en/latest/user/quickstart/#passing-parameters-in-urls. Please use this instead.

**Comment by dhruvmanila** [comment]
Oh, already merged.

**Pull request merged**